src/Stream/Reader.h

```diff
@@ -85,5 +85,26 @@ namespace Stream
 			container.resize(containerSize);
 			Read(container);
 		}
+
+		// Read characters into a string until a null terminator is encountered
+		// Does not include the null terminator in the returned string
+		// Use maxCount to restrict the size of the returned string
+		std::string ReadNullTerminatedString(std::size_t maxCount = SIZE_MAX)
```

Comment:
I think maxCount is a good idea here, but could discuss removing as well.

Reply:
Heh, I did actually stop to think about that for a bit. It makes it seem almost too much like the fixed length field reading code. Though it's not. The stream position here is only advanced up to and consuming the null byte, not up to the maximum length of the read.

I think the parameter is fine, particularly given the default selected.

Reply:
Thanks for looking it over.